src/main/java/org/apache/commons/csv/ExtendedBufferedReader.java

```diff
@@ -88,11 +89,13 @@ public int read(final char[] buf, final int offset, final int length) throws IOE
 
             for (int i = offset; i < offset + len; i++) {
                 final char ch = buf[i];
+                final int previous = i > 0 ? buf[i - 1] : lastChar;
                 if (ch == LF) {
-                    if (CR != (i > 0 ? buf[i - 1] : lastChar)) {
+                    if (CR != previous) {
                         eolCounter++;
                     }
-                } else if (ch == CR) {
+                } else if ((ch == CR)
+                        || (ch == END_OF_STREAM && previous != LF && previous != CR)) {
```

Comment:
This looks wrong as `ch == END_OF_STREAM` will always be false since a `char` cannot be negative. If you remove `ch == END_OF_STREAM &&` from the line, the test still passes. Either this code is wrong or the test does not cover the intent of this test.

Reply:
thanks ，you are right，that code is useless. I have remove it.